mongo/errors.go

```diff
@@ -112,6 +113,22 @@ func (e MongocryptdError) Unwrap() error {
 	return e.Wrapped
 }
 
+// ServerError is the interface implemented by errors returned from the server. Custom implementations of this
+// interface should not be used in production.
+type ServerError interface {
```

Comment:
Each function in the interface should have documentation. We want to eventually get to the point where users are primarily relying on this interface and don't have to check for specific types at all, so having the docs in one place would be good. The docs can be duplicated for each type that implements the interface or the docs for those functions can just say "<name> implements the ServerError interface". I'm fine with either.